[dynamo] Added cuda and triton versions to dynamo_compile

jovianjaison opened the pull request:
Summary: Add cuda and triton versions to dynamo_compile logging site.

Test Plan:
$ buck2 run mode/opt //scripts/oulgen:runner
File changed: fbcode//caffe2/torch/_dynamo/convert_frame.py
Buck UI: https://www.internalfb.com/buck2/1a8ada1f-d54e-44b2-a368-b2ff2030e113
Network: Up: 65KiB  Down: 0B  (reSessionID-8f4d1d6d-a680-4ecc-8e73-c29c932d824b)
Jobs completed: 2166. Time elapsed: 7.0s.
Cache hits: 0%. Commands: 3 (cached: 0, remote: 0, local: 3)
BUILD SUCCEEDED
...
Cuda: 12.4.0
Triton: 3.0.0

Differential Revision: D66181508




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

masnesral commented on torch/_dynamo/convert_frame.py:
```diff
@@ -25,7 +25,7 @@
 from typing_extensions import ParamSpec
 from weakref import ReferenceType
-
+import triton
```
Sometimes triton isn't available actually. You can grep around to see how we import it elsewhere, But also: probably we can put the population of these fields in a common place: https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/utils.py#L965

jovianjaison replied:
done

masnesral reviewed (changes requested):
probably can put this at the point where we actually create the final log. I added a link

masnesral commented:
Oh you may also have test failures in dynamo/test_utils.py. You'll probably just need to zero out these fields because I suspect these aren't stable, right? https://github.com/pytorch/pytorch/blob/main/test/dynamo/test_utils.py#L198

masnesral commented on torch/_dynamo/convert_frame.py:
```diff
@@ -1159,6 +1161,8 @@ def clean_for_json(d: Dict[str, Any]) -> Dict[str, Any]:
                 "structured_logging_overhead_us": to_int_us(
                     structured_logging_overhead_s
                 ),
+                "cuda_version": torch.version.cuda,
```
My comment was buried in another so maybe you missed it (or maybe I missed your response?. I think it makes more sense to put the logging here: https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/utils.py#L965

(this is common to multiple places where we do the logging, so if we put it here then all scuba logs will have these fields)

jovianjaison replied:
Yeah @masnesral, I'll them here then.

jovianjaison replied:
Added at correct location.

jovianjaison commented:
> probably can put this at the point where we actually create the final log. I added a link

updated.

masnesral commented on torch/_dynamo/convert_frame.py:
```diff
@@ -135,7 +134,6 @@
 except ModuleNotFoundError:
     np = None
 
-
```
Looks like the changes to this fille can just be reverted

jovianjaison replied:
updated

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -978,6 +980,8 @@ def us_to_ms(field):
         "remote_fx_graph_cache_put_time_ms": us_to_ms(
             "remote_fx_graph_cache_put_time_us"
         ),
+        "cuda_version": torch.version.cuda,
```
nit: See comments above. Eventually we'd plan to remove most of these fields. Mind moving this up under "inductor_config" so there's no future confusion?

jovianjaison replied:
Done.

masnesral reviewed (approved):
thanks for the changes!

jovianjaison committed:
[dynamo] Added cuda and triton versions to dynamo_compile (#141140)

Summary:

Add cuda and triton versions to dynamo_compile logging site.

Test Plan:
$ buck2 run mode/opt //scripts/oulgen:runner
File changed: fbcode//caffe2/torch/_dynamo/convert_frame.py
Buck UI: https://www.internalfb.com/buck2/1a8ada1f-d54e-44b2-a368-b2ff2030e113
Network: Up: 65KiB  Down: 0B  (reSessionID-8f4d1d6d-a680-4ecc-8e73-c29c932d824b)
Jobs completed: 2166. Time elapsed: 7.0s.
Cache hits: 0%. Commands: 3 (cached: 0, remote: 0, local: 3)
BUILD SUCCEEDED
...
Cuda: 12.4.0
Triton: 3.0.0

Reviewed By: masnesral

Differential Revision: D66181508
